diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1497055..88ec5ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ if ("$ENV{IGNITION_VERSION}" STREQUAL "fortress")
 
   message(STATUS "Compiling against Ignition Fortress")
 # Edifice
-elseif ("$ENV{IGNITION_VERSION}" STREQUAL "edifice")
+elseif ("$ENV{IGNITION_VERSION}" STREQUAL "edifice" OR TRUE)
   find_package(ignition-transport10 REQUIRED)
   set(IGN_TRANSPORT_VER ${ignition-transport10_VERSION_MAJOR})
 
