diff --git a/include/gazebo_plugins/gazebo_ros_camera_utils.h b/gazebo_plugins/include/gazebo_plugins/gazebo_ros_camera_utils.h
index eb8132d9..7f6e7044 100644
--- a/include/gazebo_plugins/gazebo_ros_camera_utils.h
+++ b/include/gazebo_plugins/gazebo_ros_camera_utils.h
@@ -47,7 +47,6 @@
 #include <gazebo/msgs/MessageTypes.hh>
 #include <gazebo/common/Time.hh>
 #include <gazebo/sensors/SensorTypes.hh>
-#include <gazebo/plugins/CameraPlugin.hh>
 #include <gazebo_plugins/gazebo_ros_utils.h>
 
 namespace gazebo

