diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed6ff0806..5679f4f3c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,10 @@ find_package(catkin REQUIRED COMPONENTS
   trajectory_msgs
 )
 
+if(MSVC)
+       add_compile_options(/permissive-)
+endif()
+
 ## System dependencies are found with CMake's conventions
 find_package(Boost REQUIRED COMPONENTS system)
 

@@ -44,7 +44,9 @@ find_package(GeographicLib REQUIRED)
 ## Check if the datasets are installed
 include(CheckGeographicLibDatasets)
 
+if(UNIX)
 include(EnableCXX11)
+endif()
 include(MavrosMavlink)
 
 # detect if sensor_msgs has BatteryState.msg

@@ -107,12 +107,17 @@ add_dependencies(mavros
   ${catkin_EXPORTED_TARGETS}
 )
 target_link_libraries(mavros
-  atomic
   ${catkin_LIBRARIES}
   ${Boost_LIBRARIES}
   ${GeographicLib_LIBRARIES}
 )
 
+if(NOT APPLE)
+  target_link_libraries(mavros
+    atomic
+  )
+endif()
+
 add_library(mavros_plugins
   src/plugins/3dr_radio.cpp
   src/plugins/actuator_control.cpp
diff --git a/include/mavros/utils.h b/include/mavros/utils.h
index c670ce2c3..144ba7d1c 100644
--- a/include/mavros/utils.h
+++ b/include/mavros/utils.h
@@ -24,6 +24,10 @@
 
 #include <ros/console.h>
 
+#ifdef PASSTHROUGH
+#undef PASSTHROUGH
+#endif
+
 // OS X compat: missing error codes
 #ifdef __APPLE__
 #define EBADE 50       /* Invalid exchange */
