diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ddedfb..2b97c7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,10 @@
 cmake_minimum_required(VERSION 3.1.0)
 project(librealsense2 LANGUAGES CXX C)
 
+if (WIN32)
+  file(COPY ${CMAKE_SOURCE_DIR}/src/win7/drivers/IntelRealSense_D400_series_win7.inf DESTINATION ${CMAKE_SOURCE_DIR})
+endif()
+
 include(CMake/lrs_options.cmake)
 include(CMake/connectivity_check.cmake)
 
diff --git a/third-party/realsense-file/CMakeLists.txt b/third-party/realsense-file/CMakeLists.txt
index becf0d0..f475e72 100644
--- a/third-party/realsense-file/CMakeLists.txt
+++ b/third-party/realsense-file/CMakeLists.txt
@@ -25,7 +25,7 @@ source_group("Header Files\\boost" REGULAR_EXPRESSION
          boost/boost/*.hpp
         )
 
-add_library(${PROJECT_NAME} STATIC
+add_library(${PROJECT_NAME}
     ${AllSources}
     ${HEADER_FILES_ROSBAG}
     ${SOURCE_FILES_ROSBAG}
