diff --git a/mapviz/CMakeLists.txt b/mapviz/CMakeLists.txt
index 943e1c2a..0a304269 100644
--- a/mapviz/CMakeLists.txt
+++ b/mapviz/CMakeLists.txt
@@ -42,7 +42,7 @@ find_package(Qt5OpenGL REQUIRED)
 find_package(Qt5Widgets REQUIRED)
 # Setting these variable so catkin_package can export Qt as a dependency
 set(Qt_FOUND TRUE)
-set(Qt_INCLUDE_DIRS "${Qt5Concurrent_INCLUDE_DIRS};${Qt5Core_INCLUDE_DIRS};${Qt5Gui_INCLUDE_DIRS};${Qt5OpenGL_INCLUDE_DIRS};${Qt5Widgets_INCLUDE_DIRS}")
+set(Qt_INCLUDE_DIRS "${Qt5Concurrent_INCLUDE_DIRS};${Qt5Core_INCLUDE_DIRS};${Qt5Gui_INCLUDE_DIRS};${Qt5Widgets_INCLUDE_DIRS}")
 set(Qt_LIBRARIES
   Qt5::Concurrent
   Qt5::Core

