diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2abf7e5..3e6ed8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,6 +75,10 @@ target_link_libraries(franka_control_node
   ${catkin_LIBRARIES}
 )
 
+if(UNIX AND NOT APPLE)
+    target_link_libraries(franka_control_node rt)
+endif()
+
 target_include_directories(franka_control_node SYSTEM PUBLIC
   ${Franka_INCLUDE_DIRS}
   ${catkin_INCLUDE_DIRS}
