diff --git a/moveit_core/CMakeLists.txt b/moveit_core/CMakeLists.txt
index e7ffb293c5..fb7666d196 100644
--- a/moveit_core/CMakeLists.txt
+++ b/moveit_core/CMakeLists.txt
@@ -8,6 +8,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 endif()
 include(cmake/moveit.cmake)
 moveit_build_options()
+set(CMAKE_CXX_STANDARD 17)
 
 find_package(Boost REQUIRED system filesystem date_time thread iostreams regex ${EXTRA_BOOST_COMPONENTS})
 find_package(Eigen3 REQUIRED)
