diff --git a/CMakeLists.txt b/CMakeLists.txt
index b353bffc7..5f5d938fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.15)
 project(moveit_core)
 
 if(NOT "${CMAKE_CXX_STANDARD}")
@@ -82,9 +82,11 @@ COMPONENTS
   urdf
   visualization_msgs
   xmlrpcpp
-  pybind11_catkin
 )
 
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
+find_package(pybind11 CONFIG REQUIRED)
+
 catkin_python_setup()
 
 set(VERSION_FILE_PATH "${CATKIN_DEVEL_PREFIX}/include")
@@ -242,7 +244,7 @@ set(pymoveit_libs
   moveit_transforms
 )
 
-pybind_add_module(pymoveit_core
+pybind11_add_module(pymoveit_core
     python/pymoveit_core.cpp
     collision_detection/src/pycollision_detection.cpp
     robot_model/src/pyrobot_model.cpp
diff --git a/collision_detection_bullet/CMakeLists.txt b/collision_detection_bullet/CMakeLists.txt
index 775fba1bd..5d209dc9e 100644
--- a/collision_detection_bullet/CMakeLists.txt
+++ b/collision_detection_bullet/CMakeLists.txt
@@ -17,7 +17,7 @@ add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
 
 add_library(collision_detector_bullet_plugin src/collision_detector_bullet_plugin_loader.cpp)
 set_target_properties(collision_detector_bullet_plugin PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(collision_detector_bullet_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME})
+target_link_libraries(collision_detector_bullet_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME} moveit_planning_scene)
 
 install(TARGETS ${MOVEIT_LIB_NAME} collision_detector_bullet_plugin
         LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
diff --git a/collision_detection_fcl/CMakeLists.txt b/collision_detection_fcl/CMakeLists.txt
index 48bb00e6e..aaefe476c 100644
--- a/collision_detection_fcl/CMakeLists.txt
+++ b/collision_detection_fcl/CMakeLists.txt
@@ -11,7 +11,7 @@ add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
 
 add_library(collision_detector_fcl_plugin src/collision_detector_fcl_plugin_loader.cpp)
 set_target_properties(collision_detector_fcl_plugin PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(collision_detector_fcl_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME})
+target_link_libraries(collision_detector_fcl_plugin ${catkin_LIBRARIES} ${MOVEIT_LIB_NAME} moveit_planning_scene)
 
 
 install(TARGETS ${MOVEIT_LIB_NAME} collision_detector_fcl_plugin
diff --git a/python/tools/CMakeLists.txt b/python/tools/CMakeLists.txt
index 6a42a03fe..101888f02 100644
--- a/python/tools/CMakeLists.txt
+++ b/python/tools/CMakeLists.txt
@@ -4,7 +4,7 @@ add_library(${MOVEIT_LIB_NAME}
   src/pybind_rosmsg_typecasters.cpp
 )
 add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(${MOVEIT_LIB_NAME} ${catkin_LIBRARIES})
+target_link_libraries(${MOVEIT_LIB_NAME} ${catkin_LIBRARIES} pybind11::module)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
 install(
