diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ccf4bc..1b2ff24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,10 +15,21 @@ catkin_package(
   CATKIN_DEPENDS sensor_msgs
 )
 
-find_package(yaml-cpp)
-add_definitions(-DHAVE_NEW_YAMLCPP)
-link_directories(${YAML_CPP_LIBRARY_DIRS})
-include_directories(${YAML_CPP_INCLUDE_DIRS})
+# find_package(yaml-cpp)
+# add_definitions(-DHAVE_NEW_YAMLCPP)
+# link_directories(${YAML_CPP_LIBRARY_DIRS})
+# include_directories(${YAML_CPP_INCLUDE_DIRS})
+if(NOT YAMLCPP_FOUND)
+  find_package(yaml-cpp 0.6 REQUIRED)
+  set(YAMLCPP_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
+  set(YAMLCPP_LIBRARIES ${YAML_CPP_LIBRARIES})
+  add_definitions(-DHAVE_NEW_YAMLCPP)
+else()
+  if(YAMLCPP_VERSION VERSION_GREATER "0.5.0")
+    add_definitions(-DHAVE_NEW_YAMLCPP)
+  endif()
+  link_directories(${YAMLCPP_LIBRARY_DIRS})
+endif()
 
 # define the library
 add_library(${PROJECT_NAME}
@@ -30,7 +41,7 @@ add_library(${PROJECT_NAME}
 add_library(${PROJECT_NAME}_wrapper
   src/parse_wrapper.cpp)
 
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${YAML_CPP_LIBRARIES} ${Boost_FILESYSTEM_LIBRARY})
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${YAMLCPP_LIBRARIES} ${Boost_FILESYSTEM_LIBRARY})
 target_link_libraries(${PROJECT_NAME}_wrapper ${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
 
 # Don't prepend wrapper library name with lib and add to Python libs.
