diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95bf11a..60b668c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(catkin REQUIRED COMPONENTS
 catkin_package()
 include_directories(${catkin_INCLUDE_DIRS})
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Werror")
+# set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Werror")
 
 if(${pluginlib_VERSION} VERSION_GREATER 1.11.1)
 	message(STATUS "using new pluginlib headers with .hpp extensions")
