diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe506f1ffc..783dae7e34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,9 +53,13 @@ if(CMAKE_THREAD_LIBS_INIT)
   endif()
 endif()
 
+if (UNIX AND NOT APPLE)
+  set(RT_LIB rt)
+endif()
+
 catkin_package(
   INCLUDE_DIRS include ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}/ros
-  LIBRARIES roscpp ${PTHREAD_LIB}
+  LIBRARIES roscpp ${PTHREAD_LIB} ${RT_LIB}
   CATKIN_DEPENDS cpp_common message_runtime rosconsole roscpp_serialization roscpp_traits rosgraph_msgs rostime std_msgs xmlrpcpp
   DEPENDS Boost
 )
