diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bcd328bc..1a5482604 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,9 @@ find_package(Boost REQUIRED COMPONENTS system)
 
 # add package modules path, not needed in dependend packages
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+if(UNIX)
 include(EnableCXX11)
+endif()
 include(MavrosMavlink)
 
 # Fixed in mavlink 2016.7.7
diff --git a/include/mavconn/msgbuffer.h b/include/mavconn/msgbuffer.h
index 539f20341..694649063 100644
--- a/include/mavconn/msgbuffer.h
+++ b/include/mavconn/msgbuffer.h
@@ -20,6 +20,11 @@
 #include <cassert>
 #include <mavconn/mavlink_dialect.h>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 namespace mavconn {
 /**
  * @brief Message buffer for internal use in libmavconn
