diff --git a/tools/rosbag/src/recorder.cpp b/tools/rosbag/src/recorder.cpp
index 88976b5ca8..1212883900 100644
--- a/tools/rosbag/src/recorder.cpp
+++ b/tools/rosbag/src/recorder.cpp
@@ -349,7 +349,7 @@ void Recorder::doQueue(const ros::MessageEvent<topic_tools::ShapeShifter const>&
                 if (it2 != out.connection_header->end())
                 {
                     auto const result = latched_msgs_.insert({{subscriber->getTopic(), it2->second}, out});
-                    if (not result.second)  // The map::insert function does not update values of existing keys
+                    if (!result.second)  // The map::insert function does not update values of existing keys
                     {
                         result.first->second = out;
                     }

