diff --git a/plotter_gui/curvelist_view.cpp b/plotter_gui/curvelist_view.cpp
index fb7aae6..9beece2 100644
--- a/plotter_gui/curvelist_view.cpp
+++ b/plotter_gui/curvelist_view.cpp
@@ -38,7 +38,7 @@ void CurveTableView::addItem(const QString& item_name)
     return;
   }
 
-  auto item = new SortedTableItem<QTableWidgetItem>(item_name);
+  auto item = new SortedTableItem(item_name);
   QFont font = QFontDatabase::systemFont(QFontDatabase::GeneralFont);
   font.setPointSize(_point_size);
   item->setFont(font);
diff --git a/plotter_gui/curvelist_view.h b/plotter_gui/curvelist_view.h
index 99d502b..9652994 100644
--- a/plotter_gui/curvelist_view.h
+++ b/plotter_gui/curvelist_view.h
@@ -18,11 +18,10 @@
 
 class CurveListPanel;
 
-template <typename ItemType>
-class SortedTableItem : public ItemType
+class SortedTableItem : public QTableWidgetItem
 {
 public:
-  SortedTableItem(const QString& name) : ItemType(name), str(name.toStdString())
+  SortedTableItem(const QString& name) : QTableWidgetItem(name), str(name.toStdString())
   {
   }
 
diff --git a/plugins/ROS/ros1_parsers/tf_msg.h b/plugins/ROS/ros1_parsers/tf_msg.h
index 978d3ab..073d53e 100644
--- a/plugins/ROS/ros1_parsers/tf_msg.h
+++ b/plugins/ROS/ros1_parsers/tf_msg.h
@@ -17,21 +17,21 @@ public:
 
   void parseMessageImpl(const TfMsgType& msg, double timestamp) override
   {
-    auto GetSeries = [&](const std::string& name) { return &BaseParser::getSeries(BaseParser::_plot_data, name); };
+    auto GetSeries = [&](const std::string& name) { return &getSeries(_plot_data, name); };
 
     for (const auto& trans : msg.transforms)
     {
       double header_stamp = trans.header.stamp.toSec();
-      timestamp = (BaseParser::_use_header_stamp && header_stamp > 0) ? header_stamp : timestamp;
+      timestamp = (_use_header_stamp && header_stamp > 0) ? header_stamp : timestamp;
 
       std::string prefix;
       if (trans.header.frame_id.empty())
       {
-        prefix = fmt::format("{}/{}", BaseParser::_topic_name, trans.child_frame_id);
+        prefix = fmt::format("{}/{}", _topic_name, trans.child_frame_id);
       }
       else
       {
-        prefix = fmt::format("{}/{}/{}", BaseParser::_topic_name, trans.header.frame_id, trans.child_frame_id);
+        prefix = fmt::format("{}/{}/{}", _topic_name, trans.header.frame_id, trans.child_frame_id);
       }
 
       PlotData* series = GetSeries(prefix + "/header/stamp");
