diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5bc1990..b12cfa5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,8 +4,8 @@ find_package(catkin REQUIRED COMPONENTS ecl_command_line)
 
 # pkg-config packages
 find_package(PkgConfig)
-pkg_check_modules(libusb REQUIRED libusb)
-pkg_check_modules(libftdi REQUIRED libftdi)
+pkg_check_modules(libusb REQUIRED libusb-1.0)
+pkg_check_modules(libftdi REQUIRED libftdi1)
 
 catkin_package(
    INCLUDE_DIRS include
