diff --git a/src/qt_gui_cpp_sip/CMakeLists.txt b/src/qt_gui_cpp_sip/CMakeLists.txt
index 370457fe..081e631d 100644
--- a/src/qt_gui_cpp_sip/CMakeLists.txt
+++ b/src/qt_gui_cpp_sip/CMakeLists.txt
@@ -26,12 +26,31 @@ set(qt_gui_cpp_sip_DEPENDENT_FILES
   ${qt_gui_cpp_HDRS_DIR}/settings.h
 )
 
+find_package(Python REQUIRED COMPONENTS Interpreter Development)
+find_package(TinyXML REQUIRED)
+
+
+
 # maintain context for different named target
 set(qt_gui_cpp_sip_INCLUDE_DIRS ${qt_gui_cpp_INCLUDE_DIRS} "${CMAKE_CURRENT_SOURCE_DIR}/../../include" ${catkin_INCLUDE_DIRS})
-set(qt_gui_cpp_sip_LIBRARIES ${qt_gui_cpp_LIBRARIES} ${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-set(qt_gui_cpp_sip_LIBRARY_DIRS ${qt_gui_cpp_LIBRARY_DIRS} ${CATKIN_DEVEL_PREFIX}/lib)
+set(_qt_gui_cpp_sip_LIBRARIES ${qt_gui_cpp_LIBRARIES} ${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${TinyXML_LIBRARIES} Python::Python)
+
+message(WARNING "CMAKE_SYSTEM_PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
+set(qt_gui_cpp_sip_LIBRARY_DIRS ${CMAKE_BINARY_DIR} ${qt_gui_cpp_LIBRARY_DIRS} ${CATKIN_DEVEL_PREFIX}/lib /usr/lib/${CMAKE_SYSTEM_PROCESSOR}-linux-gnu/)
 set(qt_gui_cpp_sip_LDFLAGS_OTHER ${qt_gui_cpp_LDFLAGS_OTHER})
 
+# sip needs libraries to have resolved paths and cannot link to cmake targets
+foreach(_lib_name_raw ${_qt_gui_cpp_sip_LIBRARIES})
+  string(REGEX REPLACE "\\.so\\.[0-9,\\.]*" ".so" _lib_name ${_lib_name_raw})
+  if(TARGET ${_lib_name})
+    # Use a nifty cmake generator expression to resolve the target location
+    list(APPEND qt_gui_cpp_sip_LIBRARIES $<TARGET_FILE:${_lib_name}>)
+  else()
+    # This library should work as is
+    list(APPEND qt_gui_cpp_sip_LIBRARIES ${_lib_name})
+  endif()
+endforeach()
+
 find_package(python_qt_binding REQUIRED)
 include(${python_qt_binding_EXTRAS_DIR}/sip_helper.cmake)
 
@@ -47,11 +66,11 @@ if(sip_helper_FOUND)
   )
 
   if(APPLE)
-    set(LIBQT_GUI_CPP_SIP_SUFFIX .so)
+    set(LIBQT_GUI_CPP_SIP_SUFFIX ".cpython-${Python_VERSION_MAJOR}${Python_VERSION_MINOR}-darwin.so")
   elseif(WIN32)
     set(LIBQT_GUI_CPP_SIP_SUFFIX .pyd)
   else()
-    set(LIBQT_GUI_CPP_SIP_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(LIBQT_GUI_CPP_SIP_SUFFIX ".cpython-${Python_VERSION_MAJOR}${Python_VERSION_MINOR}-${CMAKE_SYSTEM_PROCESSOR}-linux-gnu${CMAKE_SHARED_LIBRARY_SUFFIX}")
   endif()
 
   install(FILES ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_PYTHON_DESTINATION}/libqt_gui_cpp_sip${LIBQT_GUI_CPP_SIP_SUFFIX}
diff --git a/src/qt_gui_cpp_sip/qt_gui_cpp.sip b/src/qt_gui_cpp_sip/qt_gui_cpp.sip
index ef3c52d7..cd02078d 100644
--- a/src/qt_gui_cpp_sip/qt_gui_cpp.sip
+++ b/src/qt_gui_cpp_sip/qt_gui_cpp.sip
@@ -1,4 +1,4 @@
-%Module libqt_gui_cpp_sip 1
+%Module libqt_gui_cpp_sip
 
 %Import QtCore/QtCoremod.sip
 %Import QtGui/QtGuimod.sip
diff --git a/src/qt_gui_cpp/CMakeLists.txt b/src/qt_gui_cpp/CMakeLists.txt
index 9e9a0b92..82a91a9d 100644
--- a/src/qt_gui_cpp/CMakeLists.txt
+++ b/src/qt_gui_cpp/CMakeLists.txt
@@ -1,6 +1,6 @@
 set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
 find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt5 COMPONENTS Core Gui Widgets REQUIRED)
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 endif()
