diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb01537..69c8aac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,8 @@ find_package(catkin REQUIRED COMPONENTS
 )
 find_package(Boost REQUIRED COMPONENTS thread)

+find_package(Threads)
+
 catkin_python_setup()

 catkin_package(
@@ -49,6 +51,9 @@ add_library(${PROJECT_NAME}
 )
 add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
 target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
+if(UNIX AND NOT APPLE)
+    target_link_libraries(${PROJECT_NAME} rt)
+endif()

 # buffer_server executable
 add_executable(${PROJECT_NAME}_buffer_server src/buffer_server_main.cpp)
@@ -57,6 +62,7 @@ target_link_libraries(${PROJECT_NAME}_buffer_server
   ${PROJECT_NAME}
   ${catkin_LIBRARIES}
   ${Boost_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
 )
 set_target_properties(${PROJECT_NAME}_buffer_server
   PROPERTIES OUTPUT_NAME buffer_server
@@ -70,6 +76,7 @@ add_dependencies(${PROJECT_NAME}_static_transform_publisher ${catkin_EXPORTED_TA
 target_link_libraries(${PROJECT_NAME}_static_transform_publisher
   ${PROJECT_NAME}
   ${catkin_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
 )
 set_target_properties(${PROJECT_NAME}_static_transform_publisher
   PROPERTIES OUTPUT_NAME static_transform_publisher
