diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56fd522..4574e58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,7 +114,6 @@ catkin_package(
     tf2_ros
   DEPENDS
     ${EIGEN_PACKAGE}
-    YAML_CPP
 )
 
 ###########
@@ -127,8 +126,6 @@ include_directories(
   ${EIGEN3_INCLUDE_DIRS}
   ${YAML_CPP_INCLUDE_DIRS})
 
-link_directories(${YAML_CPP_LIBRARY_DIRS})
-
 # Library definitions
 add_library(filter_utilities src/filter_utilities.cpp)
 add_library(filter_base src/filter_base.cpp)
