diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8e8f70..fa1f242 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
 add_library(catch_ros_standalone
 	src/standalone_main.cpp
+	src/meta_info.cpp
 )
 target_link_libraries(catch_ros_standalone
 	${Boost_LIBRARIES}
@@ -27,9 +28,11 @@ target_link_libraries(catch_ros_standalone
 
 add_library(catch_ros_rostest
 	src/rostest_main.cpp
+	src/meta_info.cpp
 )
 target_link_libraries(catch_ros_rostest
 	${Boost_LIBRARIES}
+	${catkin_LIBRARIES}
 )
 
 install(FILES src/meta_info.cpp DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
