diff --git a/CMakeLists.txt b/CMakeLists.txt
index bd600b66..ba8780ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,8 +20,8 @@ set(CMAKE_INSTALL_LIBDIR lib)
 if (DEFINED ENV{ROS_PYTHON_VERSION})
     set(Py_VERSION $ENV{ROS_PYTHON_VERSION})
 else()
-    message(WARNING "ENV ROS_PYTHON_VERSION is not set. Use default: 2")
-    set(Py_VERSION "2")
+    message(WARNING "ENV ROS_PYTHON_VERSION is not set. Use default: 3")
+    set(Py_VERSION "3")
 endif()
 
 if (NOT CMAKE_VERSION VERSION_LESS "3.12")


diff --git a/pymavlink/generator/CPP11/include_v2.0/msgmap.hpp b/pymavlink/generator/CPP11/include_v2.0/msgmap.hpp
index 92cf9e7a..a3956aa2 100644
--- a/pymavlink/generator/CPP11/include_v2.0/msgmap.hpp
+++ b/pymavlink/generator/CPP11/include_v2.0/msgmap.hpp
@@ -2,7 +2,13 @@
 #pragma once
 
 #include <algorithm>
+#ifdef FREEBSD
+#include <sys/endian.h>
+#elif __APPLE__
+#include <machine/endian.h>
+#else
 #include <endian.h>
+#endif
 #include <type_traits>
 
 namespace mavlink {
