diff --git a/diagnostic_updater/CMakeLists.txt b/diagnostic_updater/CMakeLists.txt
index add78c3f..0cfe20cd 100644
--- a/diagnostic_updater/CMakeLists.txt
+++ b/diagnostic_updater/CMakeLists.txt
@@ -6,7 +6,7 @@ find_package(catkin REQUIRED diagnostic_msgs roscpp std_msgs)
 
 catkin_python_setup()
 
-catkin_package(DEPENDS diagnostic_msgs roscpp std_msgs
+catkin_package(CATKIN_DEPENDS diagnostic_msgs roscpp std_msgs
                INCLUDE_DIRS include
 	       LIBRARIES ${PROJECT_NAME}
 )
@@ -15,6 +15,7 @@ include_directories(include ${catkin_INCLUDE_DIRS})
 add_executable(example src/example.cpp)
 add_library(${PROJECT_NAME}
   src/timestamp_status.cpp)
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
 target_link_libraries(example ${catkin_LIBRARIES})
 
 #set(LOCAL_GTEST_DIR "gtest-1.7.0")

diff --git a/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.h b/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.h
index cd9178ee..5fa27422 100644
--- a/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.h
+++ b/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.h
@@ -43,6 +43,7 @@
 #include "ros/this_node.h"
 
 #include "diagnostic_msgs/DiagnosticArray.h"
+#include "diagnostic_msgs/DiagnosticStatus.h"
 #include "diagnostic_updater/DiagnosticStatusWrapper.h"
 
 #include <boost/thread.hpp>

