diff --git a/pick_place/include/moveit/pick_place/pick_place.h b/pick_place/include/moveit/pick_place/pick_place.h
index 6c2118562..ec804cab5 100644
--- a/pick_place/include/moveit/pick_place/pick_place.h
+++ b/pick_place/include/moveit/pick_place/pick_place.h
@@ -46,6 +46,18 @@
 #include <boost/noncopyable.hpp>
 #include <memory>
 
+// Import/export for windows dll's and visibility for gcc shared libraries.
+
+#ifdef ROS_BUILD_SHARED_LIBS // ros is being built around shared libraries
+  #ifdef moveit_pick_place_planner_EXPORTS // we are building a shared lib/dll
+    #define MOVEIT_PICK_PLACE_DECL ROS_HELPER_EXPORT
+  #else // we are using shared lib/dll
+    #define MOVEIT_PICK_PLACE_DECL ROS_HELPER_IMPORT
+  #endif
+#else // ros is being built around static libraries
+  #define MOVEIT_PICK_PLACE_DECL
+#endif
+
 namespace pick_place
 {
 MOVEIT_CLASS_FORWARD(PickPlace);
@@ -113,8 +125,8 @@ public:
 class PickPlace : private boost::noncopyable, public std::enable_shared_from_this<PickPlace>
 {
 public:
-  static const std::string DISPLAY_PATH_TOPIC;
-  static const std::string DISPLAY_GRASP_TOPIC;
+  static MOVEIT_PICK_PLACE_DECL const std::string DISPLAY_PATH_TOPIC;
+  static MOVEIT_PICK_PLACE_DECL const std::string DISPLAY_GRASP_TOPIC;
 
   // the amount of time (maximum) to wait for achieving a grasp posture
   static const double DEFAULT_GRASP_POSTURE_COMPLETION_DURATION;  // seconds
