diff --git a/cmake/compiler.cmake b/cmake/compiler.cmake
index a191c80..c187bed 100644
--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -31,9 +31,9 @@ MACRO(_SETUP_PROJECT_WARNINGS)
     ENDIF(${R${FLAG}})
   ENDFOREACH(FLAG ${FLAGS})
 
-  IF(NOT DEFINED CXX_DISABLE_WERROR)
-    SET(WARNING_CXX_FLAGS "-Werror ${WARNING_CXX_FLAGS}")
-  ENDIF(NOT DEFINED CXX_DISABLE_WERROR)
+  # IF(NOT DEFINED CXX_DISABLE_WERROR)
+  #   SET(WARNING_CXX_FLAGS "-Werror ${WARNING_CXX_FLAGS}")
+  # ENDIF(NOT DEFINED CXX_DISABLE_WERROR)
  ENDIF(UNIX)
 
  # For win32 systems, it is impossible to use Wall,
