diff --git a/CMakeLists.txt b/CMakeLists.txt
index 674e1033..7deeb26d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,12 +25,7 @@ find_package(Boost REQUIRED
 
 find_package(urdfdom_headers REQUIRED)
 
-set(CMAKE_POLICY_DEFAULT_CMP0012 NEW)  # required due to a bug in assimp 5.0
-find_package(ASSIMP REQUIRED)
-if(NOT DEFINED ASSIMP_LIBRARIES AND TARGET assimp::assimp)
-  # ASSIMP >= 5.0 no longer defines ASSIMP_LIBRARIES
-  set(ASSIMP_LIBRARIES assimp::assimp)
-endif()
+find_package(assimp REQUIRED)
 
 find_package(OGRE QUIET COMPONENTS Overlay)
 if(NOT OGRE_FOUND)
@@ -97,10 +92,10 @@ if(APPLE)
 endif()
 
 # Prefer newer vender-specific OpenGL library
-if (POLICY CMP0072)
-  cmake_policy(SET CMP0072 NEW)
-endif()
-find_package(OpenGL REQUIRED)
+# if (POLICY CMP0072)
+#   cmake_policy(SET CMP0072 NEW)
+# endif()
+# find_package(OpenGL REQUIRED)
 
 set(CMAKE_AUTOMOC ON)
 
@@ -214,7 +209,7 @@ catkin_package(
 
 #catkin_lint: ignore_once external_directory
 include_directories(src ${EXPORT_HEADER_DIR})
-include_directories(SYSTEM
+include_directories(
   ${EIGEN3_INCLUDE_DIRS}
   ${OGRE_INCLUDE_DIRS}
   ${OPENGL_INCLUDE_DIR}
diff --git a/src/python_bindings/CMakeLists.txt b/src/python_bindings/CMakeLists.txt
index aedb7ad0..1f5d1a08 100644
--- a/src/python_bindings/CMakeLists.txt
+++ b/src/python_bindings/CMakeLists.txt
@@ -2,9 +2,9 @@ set(rviz_BINDINGS "")
 
 # TODO(wjwwood): re-enabled PySide2 support when it is fixed.
 # add_subdirectory(shiboken)
-add_subdirectory(sip)
+# add_subdirectory(sip)
 
 message(STATUS "Python binding generators: ${rviz_BINDINGS}")
-if(NOT rviz_BINDINGS)
-  message(FATAL_ERROR "No Python binding generator found.")
-endif()
+# if(NOT rviz_BINDINGS)
+#   message(FATAL_ERROR "No Python binding generator found.")
+# endif()
diff --git a/src/rviz/CMakeLists.txt b/src/rviz/CMakeLists.txt
index d5941099..411512b0 100644
--- a/src/rviz/CMakeLists.txt
+++ b/src/rviz/CMakeLists.txt
@@ -145,6 +145,10 @@ target_link_libraries(${PROJECT_NAME}
   ${YAML_CPP_LIBRARIES}
 )
 
+if(UNIX AND NOT APPLE)
+    target_link_libraries(${PROJECT_NAME} rt)
+endif()
+
 add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 
diff --git a/src/rviz/ogre_helpers/render_system.cpp b/src/rviz/ogre_helpers/render_system.cpp
index d8de5b12..9499b30e 100644
--- a/src/rviz/ogre_helpers/render_system.cpp
+++ b/src/rviz/ogre_helpers/render_system.cpp
@@ -155,10 +155,7 @@ void RenderSystem::setupDummyWindowId()
 
 void RenderSystem::loadOgrePlugins()
 {
-  std::string plugin_prefix = get_ogre_plugin_path() + "/";
-#ifdef Q_OS_MAC
-  plugin_prefix += "lib";
-#endif
+  std::string plugin_prefix = std::string(std::getenv("CONDA_PREFIX")) + "/lib/OGRE/";
   ogre_root_->loadPlugin(plugin_prefix + "RenderSystem_GL");
   ogre_root_->loadPlugin(plugin_prefix + "Plugin_OctreeSceneManager");
   ogre_root_->loadPlugin(plugin_prefix + "Plugin_ParticleFX");
