diff --git a/src/aerialmap_display.cpp b/src/aerialmap_display.cpp
index 58f25b5..827b198 100644
--- a/src/aerialmap_display.cpp
+++ b/src/aerialmap_display.cpp
@@ -17,6 +17,7 @@ limitations under the License. */
 #include <OGRE/OgreSceneManager.h>
 #include <OGRE/OgreSceneNode.h>
 #include <OGRE/OgreTextureManager.h>
+#include <OGRE/OgreTechnique.h>
 
 #include "rviz/display_context.h"
 #include "rviz/frame_manager.h"
