diff --git a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_collision_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_collision_traversal_node.h
index b1224594..54200820 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_collision_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_collision_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREECOLLISIONTRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_mesh_collision_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_mesh_collision_traversal_node.h
index 304a9cb1..a5b904c8 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_mesh_collision_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_mesh_collision_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREEMESHCOLLISIONTRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_shape_collision_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_shape_collision_traversal_node.h
index 93b53abf..4668dd16 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/collision/octree_shape_collision_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/collision/octree_shape_collision_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREESHAPECOLLISIONTRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/collision/shape_octree_collision_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/collision/shape_octree_collision_traversal_node.h
index 147de063..c08ddccb 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/collision/shape_octree_collision_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/collision/shape_octree_collision_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_SHAPEOCTREECOLLISIONTRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_distance_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_distance_traversal_node.h
index 2df84244..fcef5cc6 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_distance_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_distance_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREEDISTANCETRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_mesh_distance_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_mesh_distance_traversal_node.h
index 5ac1cc28..c1a8cc2e 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_mesh_distance_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_mesh_distance_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREEMESHDISTANCETRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_shape_distance_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_shape_distance_traversal_node.h
index 482f5f87..bf7abd96 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/distance/octree_shape_distance_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/distance/octree_shape_distance_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREESHAPEDISTANCETRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/distance/shape_octree_distance_traversal_node.h b/include/fcl/narrowphase/detail/traversal/octree/distance/shape_octree_distance_traversal_node.h
index 02b3547b..8b731e16 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/distance/shape_octree_distance_traversal_node.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/distance/shape_octree_distance_traversal_node.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_SHAPEOCTREEDISTANCETRAVERSALNODE_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
diff --git a/include/fcl/narrowphase/detail/traversal/octree/octree_solver.h b/include/fcl/narrowphase/detail/traversal/octree/octree_solver.h
index 41597425..a10d835f 100644
--- a/include/fcl/narrowphase/detail/traversal/octree/octree_solver.h
+++ b/include/fcl/narrowphase/detail/traversal/octree/octree_solver.h
@@ -39,7 +39,7 @@
 #define FCL_TRAVERSAL_OCTREE_OCTREESOLVER_H
 
 #include "fcl/config.h"
-#if not(FCL_HAVE_OCTOMAP)
+#if !(FCL_HAVE_OCTOMAP)
 #error "This header requires fcl to be compiled with octomap support"
 #endif
 
