diff --git a/slam_toolbox/CMakeLists.txt b/slam_toolbox/CMakeLists.txt
index ed595f3..4879029 100644
--- a/slam_toolbox/CMakeLists.txt
+++ b/slam_toolbox/CMakeLists.txt
@@ -43,24 +43,12 @@ find_package(rostest REQUIRED)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 add_definitions(-DQT_NO_KEYWORDS)
-find_package(Boost REQUIRED system serialization filesystem thread)
-
-include_directories(include ${catkin_INCLUDE_DIRS} 
-                            ${EIGEN3_INCLUDE_DIRS} 
-                            ${CHOLMOD_INCLUDE_DIR}
-                            ${Boost_INCLUDE_DIR}
-                            ${BOOST_INCLUDE_DIRS}
-                            ${TBB_INCLUDE_DIRS}
-)
-
 add_definitions(${EIGEN3_DEFINITIONS})
+find_package(Boost REQUIRED system serialization filesystem thread)
 
 catkin_package(
     INCLUDE_DIRS
       include
-      ${BOOST_INCLUDE_DIRS}
-      ${EIGEN3_INCLUDE_DIRS}
-      ${TBB_INCLUDE_DIRS}
     LIBRARIES
       ceres_solver_plugin
       toolbox_lib
@@ -86,15 +74,31 @@ catkin_package(
       Boost
 )
 
+
+include_directories(include)
+include_directories(SYSTEM  ${catkin_INCLUDE_DIRS} 
+                            ${EIGEN3_INCLUDE_DIRS} 
+                            ${CHOLMOD_INCLUDE_DIR}
+                            ${Boost_INCLUDE_DIRS}
+                            ${BOOST_INCLUDE_DIRS}
+                            ${TBB_INCLUDE_DIRS}
+)
+
+
 #### Ceres Plugin
 add_library(ceres_solver_plugin solvers/ceres_solver.cpp)
 target_link_libraries(ceres_solver_plugin ${catkin_LIBRARIES} 
-                                          ${CERES_INCLUDE_DIRS}
                                           ${CERES_LIBRARIES}
                                           ${Boost_LIBRARIES}
                                           ${TBB_LIBRARIES}
 )
 
+get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+foreach(dir ${dirs})
+  message(STATUS "dir='${dir}'")
+endforeach()
+
+
 #### Tool lib for mapping
 add_library(toolbox_common src/slam_toolbox_common.cpp src/map_saver.cpp src/loop_closure_assistant.cpp src/laser_utils.cpp src/slam_mapper.cpp)
 target_link_libraries(toolbox_common karto ${catkin_LIBRARIES} ${Boost_LIBRARIES})
