diff --git a/moveit_ros/move_group/include/moveit/move_group/move_group_context.h b/moveit_ros/move_group/include/moveit/move_group/move_group_context.h
index e500cf5f174..e31455b331f 100644
--- a/moveit_ros/move_group/include/moveit/move_group/move_group_context.h
+++ b/moveit_ros/move_group/include/moveit/move_group/move_group_context.h
@@ -36,6 +36,7 @@
 
 #pragma once
 
+#include <string>
 #include <moveit/macros/class_forward.h>
 
 namespace moveit_cpp

