diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73f1aa3ea4..67528cf75c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,9 @@ find_package(catkin REQUIRED COMPONENTS
 )
 moveit_build_options()
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(EIGENPY REQUIRED eigenpy)
+
 find_package(PythonInterp REQUIRED)
 find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
 
@@ -77,8 +80,11 @@ include_directories(SYSTEM
                     ${catkin_INCLUDE_DIRS}
                     ${Boost_INCLUDE_DIRS}
                     ${EIGEN3_INCLUDE_DIRS}
+                    ${EIGENPY_INCLUDE_DIRS}
                     ${PYTHON_INCLUDE_DIRS})
 
+link_directories(${EIGENPY_LIBRARY_DIRS})
+
 add_subdirectory(py_bindings_tools)
 add_subdirectory(common_planning_interface_objects)
 add_subdirectory(planning_scene_interface)
diff --git a/move_group_interface/CMakeLists.txt b/move_group_interface/CMakeLists.txt
index d822174bf0..c3eb1686c0 100644
--- a/move_group_interface/CMakeLists.txt
+++ b/move_group_interface/CMakeLists.txt
@@ -2,7 +2,7 @@ set(MOVEIT_LIB_NAME moveit_move_group_interface)
 
 add_library(${MOVEIT_LIB_NAME} src/move_group_interface.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(${MOVEIT_LIB_NAME} moveit_common_planning_interface_objects moveit_planning_scene_interface ${catkin_LIBRARIES} ${Boost_LIBRARIES})
+target_link_libraries(${MOVEIT_LIB_NAME} moveit_common_planning_interface_objects moveit_planning_scene_interface ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${EIGENPY_LIBRARIES})
 add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
 
 add_library(${MOVEIT_LIB_NAME}_python src/wrap_python_move_group.cpp)
diff --git a/move_group_interface/include/moveit/move_group_interface/move_group_interface.h b/move_group_interface/include/moveit/move_group_interface/move_group_interface.h
index 17de157197..fdae58ab37 100644
--- a/move_group_interface/include/moveit/move_group_interface/move_group_interface.h
+++ b/move_group_interface/include/moveit/move_group_interface/move_group_interface.h
@@ -56,6 +56,20 @@
 #include <utility>
 #include <tf2_ros/buffer.h>
 
+#include <ros/macros.h>
+
+// Import/export for windows dll's and visibility for gcc shared libraries.
+
+#ifdef ROS_BUILD_SHARED_LIBS // ros is being built around shared libraries
+  #ifdef moveit_move_group_interface_EXPORTS // we are building a shared lib/dll
+    #define MOVEIT_MOVE_GROUP_INTERFACE_DECL ROS_HELPER_EXPORT
+  #else // we are using shared lib/dll
+    #define MOVEIT_MOVE_GROUP_INTERFACE_DECL ROS_HELPER_IMPORT
+  #endif
+#else // ros is being built around static libraries
+  #define MOVEIT_MOVE_GROUP_INTERFACE_DECL
+#endif
+
 namespace moveit
 {
 /** \brief Simple interface to MoveIt components */
@@ -70,7 +84,7 @@ MOVEIT_CLASS_FORWARD(MoveGroupInterface);  // Defines MoveGroupInterfacePtr, Con
     \brief Client class to conveniently use the ROS interfaces provided by the move_group node.
 
     This class includes many default settings to make things easy to use. */
-class MoveGroupInterface
+class MOVEIT_MOVE_GROUP_INTERFACE_DECL MoveGroupInterface
 {
 public:
   /** \brief Default ROS parameter name from where to read the robot's URDF. Set to 'robot_description' */
diff --git a/py_bindings_tools/CMakeLists.txt b/py_bindings_tools/CMakeLists.txt
index 1f82d7e4dc..9ef98d8136 100644
--- a/py_bindings_tools/CMakeLists.txt
+++ b/py_bindings_tools/CMakeLists.txt
@@ -1,5 +1,9 @@
 set(MOVEIT_LIB_NAME moveit_py_bindings_tools)
 
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" REQUIRED)
+
+
 add_library(${MOVEIT_LIB_NAME} src/roscpp_initializer.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 target_link_libraries(${MOVEIT_LIB_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
