diff --git a/cmake/python.cmake b/cmake/python.cmake
index aa5c452..4b976a1 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -620,30 +620,9 @@ endmacro()
 macro(FIND_NUMPY)
   # Detect numpy.
   message(STATUS "Checking for NumPy")
-  execute_process(
-    COMMAND "${PYTHON_EXECUTABLE}" "-c" "import numpy; print (True)"
-    OUTPUT_VARIABLE IS_NUMPY
-    ERROR_QUIET)
-  if(NOT IS_NUMPY)
-    message(FATAL_ERROR "Failed to detect numpy")
-  else()
-    if(NOT NUMPY_INCLUDE_DIRS)
-      execute_process(
-        COMMAND "${PYTHON_EXECUTABLE}" "-c"
-                "import numpy; print (numpy.get_include())"
-        OUTPUT_VARIABLE NUMPY_INCLUDE_DIRS
-        ERROR_QUIET)
-      string(REGEX REPLACE "\n$" "" NUMPY_INCLUDE_DIRS "${NUMPY_INCLUDE_DIRS}")
-      file(TO_CMAKE_PATH "${NUMPY_INCLUDE_DIRS}" NUMPY_INCLUDE_DIRS)
-    endif()
-    message(STATUS "  NUMPY_INCLUDE_DIRS=${NUMPY_INCLUDE_DIRS}")
-    # Retrive NUMPY_VERSION
-    execute_process(
-      COMMAND "${PYTHON_EXECUTABLE}" "-c"
-              "import numpy; print (numpy.__version__)"
-      OUTPUT_VARIABLE NUMPY_VERSION
-      ERROR_QUIET)
-    string(REGEX REPLACE "\n$" "" NUMPY_VERSION "${NUMPY_VERSION}")
-    message(STATUS "  NUMPY_VERSION=${NUMPY_VERSION}")
-  endif()
+  find_package (Python COMPONENTS Interpreter Development NumPy REQUIRED)
+  set(NUMPY_INCLUDE_DIRS ${Python_NumPy_INCLUDE_DIRS})
+  set(NUMPY_VERSION ${Python_NumPy_VERSION})
+  message(STATUS "  NUMPY_INCLUDE_DIRS=${NUMPY_INCLUDE_DIRS}")
+  message(STATUS "  NUMPY_VERSION=${NUMPY_VERSION}")
 endmacro()
