diff --git a/CMakeLists.txt b/CMakeLists.txt
index 15046db..c0e53dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,15 +25,15 @@ find_package(catkin REQUIRED )
 # CATKIN_MIGRATION: removed during catkin migration
 # rosbuild_init()
 
-catkin_download_test_data(download_data_willow-willow-full-0.025.pgm http://download.ros.org/data/maps/willow-full-0.025.pgm FILENAME willow-full-0.025.pgm MD5 e66b17ee374f2d7657972efcb3e2e4f7)
-catkin_download_test_data(download_data_willow-willow-full-0.05.pgm http://download.ros.org/data/maps/willow-full-0.05.pgm FILENAME willow-full-0.05.pgm  MD5 b61694296e08965096c5e78611fd9765)
-catkin_download_test_data(download_data_willow-willow-full-0.10.pgm http://download.ros.org/data/maps/willow-full-0.10.pgm FILENAME willow-full-0.10.pgm MD5 b84465cdbbfe3e2fb9eb4579e0bcaf0e)
-catkin_download_test_data(download_data_willow-willow-sans-whitelab-0.025.pgm http://download.ros.org/data/maps/willow-sans-whitelab-0.025.pgm FILENAME willow-sans-whitelab-0.025.pgm  MD5 dce0081e83a9e2eb8b29c51c8cd3ffac)
-catkin_download_test_data(download_data_willow-willow.tmap http://download.ros.org/data/maps/willow.tmap FILENAME willow.tmap MD5 48864073990c66173dade2bfcae5669d)
+catkin_download(download_data_willow-willow-full-0.025.pgm http://download.ros.org/data/maps/willow-full-0.025.pgm FILENAME willow-full-0.025.pgm MD5 e66b17ee374f2d7657972efcb3e2e4f7 DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+catkin_download(download_data_willow-willow-full-0.05.pgm http://download.ros.org/data/maps/willow-full-0.05.pgm FILENAME willow-full-0.05.pgm  MD5 b61694296e08965096c5e78611fd9765 DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+catkin_download(download_data_willow-willow-full-0.10.pgm http://download.ros.org/data/maps/willow-full-0.10.pgm FILENAME willow-full-0.10.pgm MD5 b84465cdbbfe3e2fb9eb4579e0bcaf0e DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+catkin_download(download_data_willow-willow-sans-whitelab-0.025.pgm http://download.ros.org/data/maps/willow-sans-whitelab-0.025.pgm FILENAME willow-sans-whitelab-0.025.pgm  MD5 dce0081e83a9e2eb8b29c51c8cd3ffac DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+catkin_download(download_data_willow-willow.tmap http://download.ros.org/data/maps/willow.tmap FILENAME willow.tmap MD5 48864073990c66173dade2bfcae5669d DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
 #catkin_download_test_data(download_data_willow-willow-2010-02-18-0.025.pgm http://download.ros.org/data/maps/willow-2010-02-18-0.025.pgm FILENAME willow-2010-02-18-0.025.pgm  MD5 1f82c191f1e4a830669de7bbc4bc989a)
 #catkin_download_test_data(download_data_willow-willow-sans-whitelab-2010-02-18-0.025.pgm http://download.ros.org/data/maps/willow-sans-whitelab-2010-02-18-0.025.pgm FILENAME willow-sans-whitelab-2010-02-18-0.025.pgm MD5 9fc85f28c98034979f10e8f6ece676bb)## Generate added messages and services with any dependencies listed here
 
-
+add_custom_target(download ALL DEPENDS download_extra_data)
 
 
 #generate_messages(
@@ -57,18 +57,3 @@ install(FILES
   willow.tmap.outlet_overrides.xml
   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
 )
-
-
-
-install(CODE "execute_process(COMMAND  make -C ${PROJECT_BINARY_DIR} tests)")
-install(FILES
-  ${PROJECT_BINARY_DIR}/willow-full-0.025.pgm
-  ${PROJECT_BINARY_DIR}/willow-full-0.05.pgm
-  ${PROJECT_BINARY_DIR}/willow-full-0.10.pgm
-  ${PROJECT_BINARY_DIR}/willow-sans-whitelab-0.025.pgm
-  ${PROJECT_BINARY_DIR}/willow.tmap
-#  ${PROJECT_BINARY_DIR}/willow-2010-02-18-0.025.pgm
-#  ${PROJECT_BINARY_DIR}/willow-sans-whitelab-2010-02-18-0.025.pgm
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-)
-
