diff --git a/ros_babel_fish/include/ros_babel_fish/message.h b/ros_babel_fish/include/ros_babel_fish/message.h
index 278378c..fb7184e 100644
--- a/ros_babel_fish/include/ros_babel_fish/message.h
+++ b/ros_babel_fish/include/ros_babel_fish/message.h
@@ -8,7 +8,9 @@
 #include "ros_babel_fish/exceptions/babel_fish_exception.h"
 #include <ros/time.h>
 
+#include <cstdint>
 #include <memory>
+#include <stddef.h>
 
 namespace ros_babel_fish
 {
