diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37fc096e..f9d574e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,14 +222,14 @@ if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/paho.mqtt.c/CMakeLists.txt")
     add_subdirectory( plugins/DataStreamMQTT )
 endif()
 
-if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/liblsl/CMakeLists.txt" AND NOT WIN32)
-    SET(LSL_ENABLE_TESTING FALSE CACHE BOOL "Build tests and run" FORCE)
-    SET(LSL_ENABLE_CPACK FALSE CACHE BOOL "Enable CPack" FORCE)
-    SET(LSL_BUILD_STATIC TRUE CACHE BOOL "Build static library" FORCE)
-    SET(LSL_BUILD_SHARED FALSE CACHE BOOL "Build shared library" FORCE)
-    add_subdirectory( 3rdparty/liblsl )
-    add_subdirectory( plugins/DataStreamLSL )
-endif()
+#if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/liblsl/CMakeLists.txt" AND NOT WIN32)
+#    SET(LSL_ENABLE_TESTING FALSE CACHE BOOL "Build tests and run" FORCE)
+#    SET(LSL_ENABLE_CPACK FALSE CACHE BOOL "Enable CPack" FORCE)
+#    SET(LSL_BUILD_STATIC TRUE CACHE BOOL "Build static library" FORCE)
+#    SET(LSL_BUILD_SHARED FALSE CACHE BOOL "Build shared library" FORCE)
+#    add_subdirectory( 3rdparty/liblsl )
+#    add_subdirectory( plugins/DataStreamLSL )
+#endif()
 
 add_subdirectory( src )
 

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 90a3aa46..c0ea9352 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,3 +1,4 @@
+add_definitions(-DQT_NO_FLOAT16_OPERATORS)
 
 include_directories( ./ ../  ../include )
 

