diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf57352..84b62d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,23 +161,3 @@ execute_process(COMMAND python3 -m site --user-site OUTPUT_VARIABLE PY_DEST)
 string(STRIP ${PY_DEST} PY_DEST)
 install(FILES ${PROJECT_BINARY_DIR}/apriltag${PY_EXT_SUFFIX} DESTINATION ${PY_DEST})
 endif (NOT Python3_NOT_FOUND AND NOT Numpy_NOT_FOUND AND PYTHONLIBS_FOUND AND BUILD_PYTHON_WRAPPER)
-
-
-# Examples
-# apriltag_demo
-add_executable(apriltag_demo example/apriltag_demo.c)
-target_link_libraries(apriltag_demo apriltag)
-
-# opencv_demo
-find_package(OpenCV COMPONENTS core imgproc videoio highgui QUIET)
-if(OpenCV_FOUND)
-    add_executable(opencv_demo example/opencv_demo.cc)
-    target_link_libraries(opencv_demo apriltag ${OpenCV_LIBRARIES})
-    set_target_properties(opencv_demo PROPERTIES CXX_STANDARD 11)
-    install(TARGETS opencv_demo RUNTIME DESTINATION bin)
-else()
-    message(WARNING "OpenCV not found: Not building demo")
-endif(OpenCV_FOUND)
-
-# install example programs
-install(TARGETS apriltag_demo RUNTIME DESTINATION bin)
