diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed6ff0806..594ca07c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,10 @@ find_package(catkin REQUIRED COMPONENTS
   trajectory_msgs
 )
 
+if(MSVC)
+	add_compile_options(/permissive-)
+endif()
+
 ## System dependencies are found with CMake's conventions
 find_package(Boost REQUIRED COMPONENTS system)
 
@@ -44,7 +48,9 @@ find_package(GeographicLib REQUIRED)
 ## Check if the datasets are installed
 include(CheckGeographicLibDatasets)
 
-include(EnableCXX11)
+if(UNIX)
+  include(EnableCXX11)
+endif()
 include(MavrosMavlink)
 
 # detect if sensor_msgs has BatteryState.msg
@@ -107,12 +113,17 @@ add_dependencies(mavros
   ${catkin_EXPORTED_TARGETS}
 )
 target_link_libraries(mavros
-  atomic
   ${catkin_LIBRARIES}
   ${Boost_LIBRARIES}
   ${GeographicLib_LIBRARIES}
 )
 
+if(UNIX AND NOT APPLE)
+  target_link_libraries(mavros
+    atomic
+  )
+endif()
+
 add_library(mavros_plugins
   src/plugins/3dr_radio.cpp
   src/plugins/actuator_control.cpp
diff --git a/include/mavros/utils.h b/include/mavros/utils.h
index c670ce2c3..4992dd04f 100644
--- a/include/mavros/utils.h
+++ b/include/mavros/utils.h
@@ -24,6 +24,10 @@
 
 #include <ros/console.h>
 
+#ifdef PASSTHROUGH
+#undef PASSTHROUGH
+#endif
+
 // OS X compat: missing error codes
 #ifdef __APPLE__
 #define EBADE 50	/* Invalid exchange */
@@ -32,6 +36,14 @@
 #define EBADSLT 55	/* Invalid slot */
 #endif
 
+// Win compat: missing error codes
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#define EBADE 50	/* Invalid exchange */
+#define EBADFD 81	/* File descriptor in bad state */
+#define EBADRQC 54	/* Invalid request code */
+#define EBADSLT 55	/* Invalid slot */
+#endif
+
 namespace mavros {
 namespace utils {
 using mavconn::utils::format;
diff --git a/src/lib/mavros.cpp b/src/lib/mavros.cpp
index 6a4ba43cf..bf837e7c6 100644
--- a/src/lib/mavros.cpp
+++ b/src/lib/mavros.cpp
@@ -14,7 +14,14 @@
 #include <ros/console.h>
 #include <mavros/mavros.h>
 #include <mavros/utils.h>
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#include "windows.h"
+#include "shlwapi.h"
+#pragma comment(lib, "shlwapi.lib")
+#else
+#include <dirent.h>
 #include <fnmatch.h>
+#endif
 
 // MAVLINK_VERSION string
 #include <mavlink/config.h>
@@ -263,6 +270,18 @@ void MavRos::plugin_route_cb(const mavlink_message_t *mmsg, const Framing framin
 
 static bool pattern_match(std::string &pattern, std::string &pl_name)
 {
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+    wchar_t wname[1024];
+    wchar_t wmask[1024];
+
+    size_t outsize;
+    mbstowcs_s(&outsize, wname, pattern.c_str(), strlen(pattern.c_str()) + 1);
+    mbstowcs_s(&outsize, wmask, pl_name.c_str(), strlen(pl_name.c_str()) + 1);
+    if (PathMatchSpecW(wname, wmask))
+        return true;
+    else
+        return false;
+#else
 	int cmp = fnmatch(pattern.c_str(), pl_name.c_str(), FNM_CASEFOLD);
 	if (cmp == 0)
 		return true;
@@ -274,6 +293,7 @@ static bool pattern_match(std::string &pattern, std::string &pl_name)
 	}
 
 	return false;
+#endif
 }
 
 /**
