diff --git a/cmake/platform/windows.cmake b/cmake/platform/windows.cmake
index ca8d77cd..88ad363e 100644
--- a/cmake/platform/windows.cmake
+++ b/cmake/platform/windows.cmake
@@ -85,13 +85,13 @@ if(WIN32)
   add_definitions(-D_USE_MATH_DEFINES)
 
   # do not define STRICT macros (minwindef.h or boost\winapi\basic_types.hpp)
-  add_definitions(-DNO_STRICT)
+  #add_definitions(-DSTRICT)
 
   # do not define min/max macros
   add_definitions(-DNOMINMAX)
 
   # do not define STRICT macros (qtgui\qwindowdefs_win.h)
-  add_definitions(-DQ_NOWINSTRICT)
+  #add_definitions(-DQ_WINSTRICT)
 
   # keep minimum windows headers inclusion
   add_definitions(-DWIN32_LEAN_AND_MEAN)
