diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bcd328bc..1a5482604 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,9 @@ find_package(Boost REQUIRED COMPONENTS system)
 
 # add package modules path, not needed in dependend packages
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+if(UNIX)
 include(EnableCXX11)
+endif()
 include(MavrosMavlink)
 
 # Fixed in mavlink 2016.7.7
diff --git a/include/mavconn/msgbuffer.h b/include/mavconn/msgbuffer.h
index 539f20341..694649063 100644
--- a/include/mavconn/msgbuffer.h
+++ b/include/mavconn/msgbuffer.h
@@ -20,6 +20,11 @@
 #include <cassert>
 #include <mavconn/mavlink_dialect.h>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 namespace mavconn {
 /**
  * @brief Message buffer for internal use in libmavconn
diff --git a/include/mavconn/thread_utils.h b/include/mavconn/thread_utils.h
index d388a1dcc..ef734373b 100644
--- a/include/mavconn/thread_utils.h
+++ b/include/mavconn/thread_utils.h
@@ -22,7 +22,47 @@
 #include <string>
 #include <cstdio>
 #include <sstream>
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+#include <windows.h>  
+const DWORD MS_VC_EXCEPTION = 0x406D1388;  
+#pragma pack(push,8)  
+typedef struct tagTHREADNAME_INFO  
+{  
+       DWORD dwType; // Must be 0x1000.  
+       LPCSTR szName; // Pointer to name (in user addr space).  
+       DWORD dwThreadID; // Thread ID (-1=caller thread).  
+       DWORD dwFlags; // Reserved for future use, must be zero.  
+ } THREADNAME_INFO;  
+#pragma pack(pop)  
+void SetThreadName(DWORD dwThreadID, const char* threadName) {  
+       THREADNAME_INFO info;  
+       info.dwType = 0x1000;  
+       info.szName = threadName;  
+       info.dwThreadID = dwThreadID;  
+       info.dwFlags = 0;  
+#pragma warning(push)  
+#pragma warning(disable: 6320 6322)  
+       __try{  
+               RaiseException(MS_VC_EXCEPTION, 0, sizeof(info) / sizeof(ULONG_PTR), (ULONG_PTR*)&info);  
+       }  
+       __except (EXCEPTION_EXECUTE_HANDLER){  
+       }  
+#pragma warning(pop)  
+}
+
+void SetThreadName( const char* threadName)
+{
+       SetThreadName(GetCurrentThreadId(),threadName);
+}
+
+void SetThreadName( std::thread* thread, const char* threadName)
+{
+       DWORD threadId = ::GetThreadId( static_cast<HANDLE>( thread->native_handle() ) );
+       SetThreadName(threadId,threadName);
+}
+#else
 #include <pthread.h>
+#endif
 
 namespace mavconn {
 namespace utils {
@@ -58,6 +98,9 @@ bool set_this_thread_name(const std::string &name, Args&& ... args)
 
 #ifdef __APPLE__
        return pthread_setname_np(new_name.c_str()) == 0;
+#elif defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__)
+       SetThreadName(new_name.c_str())
+       return 0;
 #else
        pthread_t pth = pthread_self();
        return pthread_setname_np(pth, new_name.c_str()) == 0;
