diff --git a/CMakeLists.txt b/CMakeLists.txt
index e67982a..4eac63d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,9 @@ endif()
 # TODO(henningkayser): Remove policy fix when assimp 5.1 is available
 # Suppress policy warning in assimp (https://github.com/assimp/assimp/pull/2722)
 set(CMAKE_POLICY_DEFAULT_CMP0012 NEW)
+set(MSVC_TOOLSET_VERSION "141")
 find_package(ASSIMP QUIET)
+unset(MSVC_TOOLSET_VERSION)
 if(NOT ASSIMP_FOUND)
   find_package(PkgConfig REQUIRED)
   # assimp is required, so REQUIRE the second attempt
@@ -24,9 +26,14 @@ endif()
 # find *absolute* paths to ASSIMP_LIBRARIES
 # Both, pkg-config and assimp's cmake-config don't provide an absolute library path.
 # For, pkg-config the path is in ASSIMP_PC_LIBRARY_DIRS, for cmake in ASSIMP_LIBRARY_DIRS.
-find_library(ASSIMP_ABS_LIBRARIES NAMES ${ASSIMP_LIBRARIES} assimp HINTS ${ASSIMP_LIBRARY_DIRS} ${ASSIMP_PC_LIBRARY_DIRS})
+find_library(ASSIMP_ABS_LIBRARIES NAMES ${ASSIMP_LIBRARIES} assimp-vc141-mt HINTS ${ASSIMP_LIBRARY_DIRS} ${ASSIMP_PC_LIBRARY_DIRS})
 set(ASSIMP_LIBRARIES "${ASSIMP_ABS_LIBRARIES}")
 
+if (WIN32)
+  # for msvc 2017 compat
+  add_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)
+endif()
+
 find_package(Boost REQUIRED system filesystem)
 
 find_package(console_bridge REQUIRED)
