diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9bd1118..b1eb309 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,18 +6,21 @@ find_package(catkin REQUIRED COMPONENTS
   std_msgs
 )
 
+pkg_check_modules(ftdi REQUIRED libftdi1)
+
 catkin_package()
 
 include_directories(
   ${catkin_INCLUDE_DIRS}
+  ${ftdi_INCLUDE_DIRS}
 )
 
 add_executable(${PROJECT_NAME}_node src/RelayNode.cpp src/node.cpp)
-target_link_libraries(${PROJECT_NAME}_node ${catkin_LIBRARIES} ftdi)
+target_link_libraries(${PROJECT_NAME}_node ${catkin_LIBRARIES} ${ftdi_LIBRARIES})
 set_target_properties(${PROJECT_NAME}_node PROPERTIES OUTPUT_NAME relay_node PREFIX "")
 
 add_executable(${PROJECT_NAME}_list src/list.cpp)
-target_link_libraries(${PROJECT_NAME}_list ${catkin_LIBRARIES} ftdi)
+target_link_libraries(${PROJECT_NAME}_list ${catkin_LIBRARIES} ${ftdi_LIBRARIES})
 set_target_properties(${PROJECT_NAME}_list PROPERTIES OUTPUT_NAME list PREFIX "")
 
 install(TARGETS ${PROJECT_NAME}_node ${PROJECT_NAME}_list
