diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37fc096e..f9d574e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,14 +222,14 @@ if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/paho.mqtt.c/CMakeLists.txt")
     add_subdirectory( plugins/DataStreamMQTT )
 endif()
 
-if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/liblsl/CMakeLists.txt")
-    SET(LSL_ENABLE_TESTING FALSE CACHE BOOL "Build tests and run" FORCE)
-    SET(LSL_ENABLE_CPACK FALSE CACHE BOOL "Enable CPack" FORCE)
-    SET(LSL_BUILD_STATIC TRUE CACHE BOOL "Build static library" FORCE)
-    SET(LSL_BUILD_SHARED FALSE CACHE BOOL "Build shared library" FORCE)
-    add_subdirectory( 3rdparty/liblsl )
-    add_subdirectory( plugins/DataStreamLSL )
-endif()
+#if(EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/liblsl/CMakeLists.txt")
+#    SET(LSL_ENABLE_TESTING FALSE CACHE BOOL "Build tests and run" FORCE)
+#    SET(LSL_ENABLE_CPACK FALSE CACHE BOOL "Enable CPack" FORCE)
+#    SET(LSL_BUILD_STATIC TRUE CACHE BOOL "Build static library" FORCE)
+#    SET(LSL_BUILD_SHARED FALSE CACHE BOOL "Build shared library" FORCE)
+#    add_subdirectory( 3rdparty/liblsl )
+#    add_subdirectory( plugins/DataStreamLSL )
+#endif()
 
 add_subdirectory( src )
 
diff --git a/include/PlotJuggler/alphanum.hpp b/include/PlotJuggler/alphanum.hpp
index 8c3594d8..9338495c 100644
--- a/include/PlotJuggler/alphanum.hpp
+++ b/include/PlotJuggler/alphanum.hpp
@@ -296,22 +296,6 @@ namespace doj
     return alphanum_impl(l, r.c_str());
   }
 
-  ////////////////////////////////////////////////////////////////////////////
-
-  /**
-     Functor class to compare two objects with the "Alphanum
-     Algorithm". If the objects are no std::string, they must
-     implement "std::ostream operator<< (std::ostream&, const Ty&)".
-  */
-  template<class Ty>
-  struct alphanum_less : public std::binary_function<Ty, Ty, bool>
-  {
-    bool operator()(const Ty& left, const Ty& right) const
-    {
-      return alphanum_comp(left, right) < 0;
-    }
-  };
-
 }
 
 

