diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ee98ee..f833789 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,9 @@ IF(NOT EXISTS "${CMAKE_SOURCE_DIR}/cmake/base.cmake")
   MESSAGE(FATAL_ERROR "\nPlease run the following command first:\ngit submodule update --init\n")
 ENDIF()
 
-set(CMAKE_BUILD_TYPE "Release")
-set(PYTHON_EXECUTABLE "/usr/bin/python3") # specify python3 so that Debian Buster passes
-set(CMAKE_INSTALL_LIBDIR "lib/")
+# set(CMAKE_BUILD_TYPE "Release")
+# set(PYTHON_EXECUTABLE "/usr/bin/python3") # specify python3 so that Debian Buster passes
+# set(CMAKE_INSTALL_LIBDIR "lib/")
 
 # Disable -Werror on Unix for now.
 SET(CXX_DISABLE_WERROR True)
