diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f9c28f4b..7390fc010 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,14 +79,6 @@ IF(MINGW)
     SET(CMAKE_SHARED_LINKER_FLAGS "-Wl,--enable-auto-import")
 ENDIF(MINGW)
 
-# GCC 4 required
-IF(UNIX OR MINGW)
-    EXEC_PROGRAM( gcc ARGS "-dumpversion" OUTPUT_VARIABLE GCC_VERSION )
-    IF(GCC_VERSION VERSION_LESS "4.0.0")
-        MESSAGE(FATAL_ERROR "GCC ${GCC_VERSION} found, but version 4.x.x minimum is required")
-    ENDIF(GCC_VERSION VERSION_LESS "4.0.0")
-ENDIF(UNIX OR MINGW)
-
 #The CDT Error Parser cannot handle error messages that span 
 #more than one line, which is the default gcc behavior. 
 #In order to force gcc to generate single line error messages with no line wrapping
@@ -162,14 +154,14 @@ OPTION(BUILD_TOOLS "Build tools" ON)
 OPTION(BUILD_EXAMPLES "Build examples" ON)
 
 ####### DEPENDENCIES #######
-IF(MOBILE_BUILD)
+IF(FALSE)
 option(WITH_QT            "Include Qt support"                   OFF)
 ELSE()
 option(WITH_QT            "Include Qt support"                   ON)
 ENDIF()
 option(WITH_ORB_OCTREE    "Include ORB Octree feature support"   ON)
 option(WITH_TORCH         "Include Torch support (SuperPoint)"   OFF)
-option(WITH_PYTHON        "Include Python3 support (PyMatcher, PyDetector)"  OFF)
+option(WITH_PYTHON        "Include Python3 support (PyMatcher, PyDetector)"  ON)
 option(WITH_PYTHON_THREADING  "Use more than one Python interpreter."  OFF)
 option(WITH_PDAL          "Include PDAL support"                 ON)
 option(WITH_FREENECT      "Include Freenect support"             ON)
@@ -348,6 +340,8 @@ IF(WITH_QT)
         ENDIF()
         ADD_DEFINITIONS(-DQT_NO_KEYWORDS) # To avoid conflicts with boost signals/foreach and Qt macros
     ENDIF(QT4_FOUND OR Qt5_FOUND)
+ELSE()
+    add_compile_definitions(DISABLE_VTK)
 ENDIF(WITH_QT)
 
 IF(NOT VTK_FOUND)
diff --git a/corelib/src/CMakeLists.txt b/corelib/src/CMakeLists.txt
index 9df6b45fd..78c02b658 100644
--- a/corelib/src/CMakeLists.txt
+++ b/corelib/src/CMakeLists.txt
@@ -159,6 +159,13 @@ SET(LIBRARIES
 	${ZLIB_LIBRARIES} 
 )
 
+IF(VTK_FOUND)
+	SET(LIBRARIES
+		${LIBRARIES}
+		${VTK_LIBRARIES}
+	)
+ENDIF()
+
 IF(Sqlite3_FOUND)
     SET(INCLUDE_DIRS
 		${INCLUDE_DIRS}
@@ -210,6 +217,8 @@ IF(WITH_PYTHON AND Python3_FOUND)
 	 SET(INCLUDE_DIRS
 	    ${TORCH_INCLUDE_DIRS}
 	    ${CMAKE_CURRENT_SOURCE_DIR}/python
+	    ${Python3_INCLUDE_DIRS}
+	    ${Python3_NumPy_INCLUDE_DIRS}
 		${INCLUDE_DIRS}
 	)
 ENDIF(WITH_PYTHON AND Python3_FOUND)
diff --git a/tools/EurocDataset/CMakeLists.txt b/tools/EurocDataset/CMakeLists.txt
index 3b1e5e4..2f23770 100644
--- a/tools/EurocDataset/CMakeLists.txt
+++ b/tools/EurocDataset/CMakeLists.txt
@@ -12,6 +12,8 @@ IF(NOT yaml-cpp_FOUND)
          SET(yaml-cpp_FOUND ${yaml_cpp_FOUND})
 	  ENDIF(yaml_cpp_FOUND)
    ENDIF(PKG_CONFIG_FOUND)
+ELSE()
+    SET(YAML_CPP_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
 ENDIF(NOT yaml-cpp_FOUND)
 
 IF(yaml-cpp_FOUND)
