diff --git a/CMakeLists.txt b/CMakeLists.txt
index fda47db..691860a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,7 @@ compute_project_args(PROJECT_ARGS LANGUAGES CXX)
 project(${PROJECT_NAME} ${PROJECT_ARGS})
 
 include("${JRL_CMAKE_MODULES}/boost.cmake")
-include("${JRL_CMAKE_MODULES}/python.cmake")
+# include("${JRL_CMAKE_MODULES}/python.cmake")
 include("${JRL_CMAKE_MODULES}/ide.cmake")
 include("${JRL_CMAKE_MODULES}/apple.cmake")
 
@@ -90,7 +90,8 @@ else()
   set(PYTHON_COMPONENTS Interpreter Development.Module NumPy)
 endif()
 set(PYTHON_EXPORT_DEPENDENCY ON)
-findpython(REQUIRED)
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs REQUIRED)
 
 if(WIN32)
   link_directories(${PYTHON_LIBRARY_DIRS})
diff --git a/cmake/_unittests/python/CMakeLists.txt b/cmake/_unittests/python/CMakeLists.txt
index 1d04e1d..5b1f75f 100644
--- a/cmake/_unittests/python/CMakeLists.txt
+++ b/cmake/_unittests/python/CMakeLists.txt
@@ -9,11 +9,13 @@ set(PROJECT_URL http://jrl-cmakemodules.readthedocs.io)
 include(../macros.cmake)
 
 include(../../base.cmake)
-include(../../python.cmake)
+# include(../../python.cmake)
 
 compute_project_args(PROJECT_ARGS)
 project(${PROJECT_NAME} ${PROJECT_ARGS})
-findpython()
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs REQUIRED)
+# findpython()
 
 check_exist_variable(PYTHON_SITELIB "")
 
