diff --git a/handeye_calibration_solver/CMakeLists.txt b/handeye_calibration_solver/CMakeLists.txt
index cab8464..90bf930 100644
--- a/handeye_calibration_solver/CMakeLists.txt
+++ b/handeye_calibration_solver/CMakeLists.txt
@@ -13,7 +13,7 @@ find_package(PythonLibs 2.7 REQUIRED)
 find_package(NumPy 1.7 REQUIRED)
 
 #catkin_lint: ignore_once missing_directory
-include_directories(${CMAKE_CURRENT_BINARY_DIR} ${PYTHON_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${PYTHON_INCLUDE_DIRS} ${NUMPY_INCLUDE_DIR})
 
 # Plugin Source
 set(SOURCE_FILES
