diff --git a/moveit_ros/robot_interaction/include/moveit/robot_interaction/kinematic_options_map.h b/moveit_ros/robot_interaction/include/moveit/robot_interaction/kinematic_options_map.h
index 40ebd87d1..8d7a8972a 100644
--- a/moveit_ros/robot_interaction/include/moveit/robot_interaction/kinematic_options_map.h
+++ b/moveit_ros/robot_interaction/include/moveit/robot_interaction/kinematic_options_map.h
@@ -40,6 +40,18 @@
 #include <boost/thread.hpp>
 #include <boost/function.hpp>
 
+// Import/export for windows dll's and visibility for gcc shared libraries.
+
+#ifdef ROS_BUILD_SHARED_LIBS // ros is being built around shared libraries
+  #ifdef moveit_robot_interaction_EXPORTS // we are building a shared lib/dll
+    #define MOVEIT_ROBOT_INTERACTION_DECL ROS_HELPER_EXPORT
+  #else // we are using shared lib/dll
+    #define MOVEIT_ROBOT_INTERACTION_DECL ROS_HELPER_IMPORT
+  #endif
+#else // ros is being built around static libraries
+  #define MOVEIT_ROBOT_INTERACTION_DECL
+#endif
+
 namespace robot_interaction
 {
 // Maintains a set of KinematicOptions with a key/value mapping and a default
@@ -51,10 +63,10 @@ public:
   KinematicOptionsMap();
 
   /// When used as \e key this means the default value
-  static const std::string DEFAULT;
+  static MOVEIT_ROBOT_INTERACTION_DECL const std::string DEFAULT;
 
   /// When used as \e key this means set ALL keys (including default)
-  static const std::string ALL;
+  static MOVEIT_ROBOT_INTERACTION_DECL const std::string ALL;
 
   /// Set \e state using inverse kinematics.
