diff --git a/CMakeLists.txt b/CMakeLists.txt
index 40d01ef..8c1fde2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,16 +8,16 @@ find_package(catkin
     roscpp
 )
 
-message(STATUS "Compiling against Ignition Citadel")
+find_package(ignition-transport9 REQUIRED)
+set(IGN_TRANSPORT_VER ${ignition-transport9_VERSION_MAJOR})
 
-find_package(ignition-gazebo3 REQUIRED)
-set(IGN_GAZEBO_VER ${ignition-gazebo3_VERSION_MAJOR})
+find_package(ignition-msgs6 REQUIRED)
+set(IGN_MSGS_VER ${ignition-msgs6_VERSION_MAJOR})
 
-find_package(ignition-transport8 REQUIRED)
-set(IGN_TRANSPORT_VER ${ignition-transport8_VERSION_MAJOR})
+find_package(ignition-gazebo4 REQUIRED)
+set(IGN_GAZEBO_VER ${ignition-gazebo4_VERSION_MAJOR})
 
-find_package(ignition-msgs5 REQUIRED)
-set(IGN_MSGS_VER ${ignition-msgs5_VERSION_MAJOR})
+message(STATUS "Compiling against Ignition Dome")
 
 ign_find_package(gflags
     REQUIRED

