diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6070e14d1..54e5f3218 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,8 +51,14 @@ else()
 endif()
 
 find_package(octomap REQUIRED)
+
 find_package(ruckig REQUIRED)
+# work around catkin_package not fetching the interface includes from the target
+# to forward to downstream dependencies. The includes do not need to be added
+# in include_directories below because the target is correctly imported here.
+get_target_property(ruckig_INCLUDE_DIRS ruckig::ruckig INTERFACE_INCLUDE_DIRECTORIES)
 set(ruckig_LIBRARIES "ruckig::ruckig")
+
 find_package(urdfdom REQUIRED)
 find_package(urdfdom_headers REQUIRED)
 
diff --git a/cmake/moveit.cmake b/cmake/moveit.cmake
index 134412ab3..83163d534 100644
--- a/cmake/moveit.cmake
+++ b/cmake/moveit.cmake
@@ -1,8 +1,9 @@
 macro(moveit_build_options)
-  # for Bionic
-  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9")
+  # for Ubuntu bionic/focal
+  if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11") OR
+     (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "16"))
     if(NOT "${CMAKE_CXX_STANDARD}")
-      set(CMAKE_CXX_STANDARD 14)
+      set(CMAKE_CXX_STANDARD 17)
     endif()
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
     set(CMAKE_CXX_EXTENSIONS OFF)
diff --git a/collision_detector_bullet_description.xml b/collision_detector_bullet_description.xml
index 6d90496ea..b7f688682 100644
--- a/collision_detector_bullet_description.xml
+++ b/collision_detector_bullet_description.xml
@@ -1,4 +1,4 @@
-<library path="lib/libcollision_detector_bullet_plugin">
+<library path="libcollision_detector_bullet_plugin">
   <class name="Bullet" type="collision_detection::CollisionDetectorBtPluginLoader"
   base_class_type="collision_detection::CollisionPlugin">
     <description>
diff --git a/collision_detector_fcl_description.xml b/collision_detector_fcl_description.xml
index f96b06c05..794b8ea2a 100644
--- a/collision_detector_fcl_description.xml
+++ b/collision_detector_fcl_description.xml
@@ -1,4 +1,4 @@
-<library path="lib/libcollision_detector_fcl_plugin">
+<library path="libcollision_detector_fcl_plugin">
   <class name="FCL" type="collision_detection::CollisionDetectorFCLPluginLoader"
   base_class_type="collision_detection::CollisionPlugin">
     <description>
diff --git a/trajectory_processing/include/moveit/trajectory_processing/ruckig_traj_smoothing.h b/trajectory_processing/include/moveit/trajectory_processing/ruckig_traj_smoothing.h
index 4671db898..d79b2645e 100644
--- a/trajectory_processing/include/moveit/trajectory_processing/ruckig_traj_smoothing.h
+++ b/trajectory_processing/include/moveit/trajectory_processing/ruckig_traj_smoothing.h
@@ -37,6 +37,7 @@
 
 #include <Eigen/Core>
 #include <list>
+#include <unordered_map>
 #include <moveit/robot_trajectory/robot_trajectory.h>
 #include <ruckig/ruckig.hpp>
 
