diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d8b818..d313db7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ generate_dynamic_reconfigure_options(
   cfg/ArucoThreshold.cfg
 )
 
-add_compile_options(-std=c++11)
+add_compile_options(-std=c++14)
 
 catkin_package(
   INCLUDE_DIRS include
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d8b818..e2c5e99 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(aruco_ros)
 
-find_package(OpenCV REQUIRED)
+find_package(OpenCV REQUIRED COMPONENTS core imgproc calib3d)
 
 find_package(catkin REQUIRED COMPONENTS
   cv_bridge
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d8b818..2738144 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,7 +35,7 @@ include_directories(
   ${OpenCV_INCLUDE_DIRS}
 )
  add_library(aruco_ros_utils src/aruco_ros_utils.cpp)
- target_link_libraries(aruco_ros_utils ${catkin_LIBRARIES})
+ target_link_libraries(aruco_ros_utils ${catkin_LIBRARIES} ${OpenCV_LIBRARIES})
 
  add_executable(single src/simple_single.cpp
                        src/aruco_ros_utils.cpp)
