diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fe8ac98..8dddf109 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,17 +185,17 @@ enable_testing()
 
 
-option(LIBNABO_BUILD_EXAMPLES "Build libnabo examples" ON)
+option(LIBNABO_BUILD_EXAMPLES "Build libnabo examples" OFF)
 if (LIBNABO_BUILD_EXAMPLES)
   add_subdirectory(examples)
 endif()
 
-option(LIBNABO_BUILD_TESTS "Build libnabo tests" ON)
+option(LIBNABO_BUILD_TESTS "Build libnabo tests" OFF)
 if(LIBNABO_BUILD_TESTS)
   add_subdirectory(tests)
 endif()
 
-option(LIBNABO_BUILD_PYTHON "Build libnabo python" ON)
+option(LIBNABO_BUILD_PYTHON "Build libnabo python" OFF)
 if(LIBNABO_BUILD_PYTHON)
   add_subdirectory(python)
 endif()
