diff --git a/CMakeLists.txt b/CMakeLists.txt
index d7d7f41..a32e8ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,23 +21,12 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
-# TODO(henningkayser): Remove policy fix when assimp 5.1 is available
-# Suppress policy warning in assimp (https://github.com/assimp/assimp/pull/2722)
-set(CMAKE_POLICY_DEFAULT_CMP0012 NEW)
-find_package(ASSIMP QUIET)
-if(NOT ASSIMP_FOUND)
-  find_package(PkgConfig REQUIRED)
-  # assimp is required, so REQUIRE the second attempt
-  pkg_check_modules(ASSIMP_PC REQUIRED assimp)
-  set(ASSIMP_INCLUDE_DIRS ${ASSIMP_PC_INCLUDE_DIRS})
+find_package(assimp REQUIRED)
+if (WIN32)
+  # for msvc 2017 compat
+  add_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)
 endif()
 
-# find *absolute* paths to ASSIMP_LIBRARIES
-# Both, pkg-config and assimp's cmake-config don't provide an absolute library path.
-# For, pkg-config the path is in ASSIMP_PC_LIBRARY_DIRS, for cmake in ASSIMP_LIBRARY_DIRS.
-find_library(ASSIMP_ABS_LIBRARIES NAMES ${ASSIMP_LIBRARIES} assimp HINTS ${ASSIMP_LIBRARY_DIRS} ${ASSIMP_PC_LIBRARY_DIRS})
-set(ASSIMP_LIBRARIES "${ASSIMP_ABS_LIBRARIES}")
-
 find_package(Boost REQUIRED system filesystem)
 
 find_package(console_bridge REQUIRED)
@@ -86,7 +75,7 @@ add_library(${PROJECT_NAME}
   src/shape_to_marker.cpp
   src/shapes.cpp
 )
-target_compile_options(${PROJECT_NAME} PRIVATE ${PROJECT_COMPILE_OPTIONS})
+# target_compile_options(${PROJECT_NAME} PRIVATE ${PROJECT_COMPILE_OPTIONS})
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
 
 target_link_libraries(${PROJECT_NAME} ${ASSIMP_LIBRARIES} ${QHULL_LIBRARIES} ${catkin_LIBRARIES} ${console_bridge_LIBRARIES} ${Boost_LIBRARIES})
