diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e37b59..086dbe6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,12 +106,14 @@ add_library(scanmatcher
   scanmatcher/scanmatcher.cpp
   scanmatcher/scanmatcherprocessor.cpp
   scanmatcher/eig3.cpp)
-add_executable(scanmatch_test
-  scanmatcher/scanmatch_test.cpp)
-add_executable(icptest
-  scanmatcher/icptest.cpp)
-target_link_libraries(scanmatch_test scanmatcher)
-target_link_libraries(icptest scanmatcher)
+if(UNIX)
+  add_executable(scanmatch_test
+    scanmatcher/scanmatch_test.cpp)
+  add_executable(icptest
+    scanmatcher/icptest.cpp)
+  target_link_libraries(scanmatch_test scanmatcher)
+  target_link_libraries(icptest scanmatcher)
+endif()
 target_link_libraries(scanmatcher
   log sensor_range sensor_odometry sensor_base utils)
 
@@ -152,12 +154,21 @@ target_link_libraries(gridfastslam
 # )
 
 ## Mark executables and/or libraries for installation
-install(TARGETS utils autoptr_test sensor_base sensor_odometry sensor_range sensor_range log log_test log_plot scanstudio2carmen rdk2carmen configfile configfile_test scanmatcher scanmatch_test icptest gridfastslam gfs2log gfs2rec gfs2neff
+install(TARGETS utils autoptr_test sensor_base sensor_odometry sensor_range sensor_range log log_test log_plot scanstudio2carmen rdk2carmen configfile configfile_test scanmatcher gridfastslam gfs2log gfs2rec gfs2neff
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
 )
 
+if(UNIX)
+  install(TARGETS scanmatch_test icptest
+    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  )
+endif()
+
+
 ## Mark cpp header files for installation
 install(DIRECTORY include/gmapping
   DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}
diff --git a/carmenwrapper/carmenwrapper.cpp b/carmenwrapper/carmenwrapper.cpp
index d51d753..f2378f2 100644
--- a/carmenwrapper/carmenwrapper.cpp
+++ b/carmenwrapper/carmenwrapper.cpp
@@ -26,6 +26,10 @@
 using namespace GMapping;
 using namespace std;
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 //static vars for the carmenwrapper
 SensorMap CarmenWrapper::m_sensorMap;
 deque<RangeReading> CarmenWrapper::m_rangeDeque;
diff --git a/gfs-carmen/gfs-carmen.cpp b/gfs-carmen/gfs-carmen.cpp
index 7c19e65..0cde8b7 100644
--- a/gfs-carmen/gfs-carmen.cpp
+++ b/gfs-carmen/gfs-carmen.cpp
@@ -27,6 +27,10 @@
 #include <gmapping/utils/orientedboundingbox.h>
 #include <gmapping/configfile/configfile.h>
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 #define DEBUG cout << __PRETTY_FUNCTION__
 
 /*
diff --git a/gridfastslam/gridslamprocessor.cpp b/gridfastslam/gridslamprocessor.cpp
index 3466746..e6c5e4a 100644
--- a/gridfastslam/gridslamprocessor.cpp
+++ b/gridfastslam/gridslamprocessor.cpp
@@ -11,6 +11,10 @@
 //#define MAP_CONSISTENCY_CHECK
 //#define GENERATE_TRAJECTORIES
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 namespace GMapping {
 
 const double m_distanceThresholdCheck = 20;
diff --git a/gui/gsp_thread.cpp b/gui/gsp_thread.cpp
index e6fb208..828dda5 100644
--- a/gui/gsp_thread.cpp
+++ b/gui/gsp_thread.cpp
@@ -30,6 +30,10 @@
 	#include <gmapping/carmenwrapper/carmenwrapper.h>
 #endif
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 #define DEBUG cout << __PRETTY_FUNCTION__
 
 using namespace std;
diff --git a/include/gmapping/grid/array2d.h b/include/gmapping/grid/array2d.h
index 2494350..6c594e0 100644
--- a/include/gmapping/grid/array2d.h
+++ b/include/gmapping/grid/array2d.h
@@ -7,6 +7,11 @@
 
 #include <iostream>
 
+
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 namespace GMapping {
 
 template<class Cell, const bool debug=false> class Array2D{
diff --git a/include/gmapping/scanmatcher/scanmatcher.h b/include/gmapping/scanmatcher/scanmatcher.h
index c970ae1..3c3e817 100644
--- a/include/gmapping/scanmatcher/scanmatcher.h
+++ b/include/gmapping/scanmatcher/scanmatcher.h
@@ -37,7 +37,7 @@ class ScanMatcher{
 		inline const double* laserAngles() const { return m_laserAngles; }
 		inline unsigned int laserBeams() const { return m_laserBeams; }
 		
-		static const double nullLikelihood;
+		double nullLikelihood=-.5;
 	protected:
 		//state of the matcher
 		bool m_activeAreaComputed;
diff --git a/include/gmapping/utils/point.h b/include/gmapping/utils/point.h
index 3735aed..153d5fb 100644
--- a/include/gmapping/utils/point.h
+++ b/include/gmapping/utils/point.h
@@ -5,6 +5,10 @@
 #include <iostream>
 #include "gmapping/utils/gvalues.h"
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 #define DEBUG_STREAM cerr << __PRETTY_FUNCTION__ << ":" //FIXME
 
 namespace GMapping {
diff --git a/include/gmapping/utils/stat.h b/include/gmapping/utils/stat.h
index 62e14fa..07ce79b 100644
--- a/include/gmapping/utils/stat.h
+++ b/include/gmapping/utils/stat.h
@@ -8,7 +8,7 @@ namespace GMapping {
 
 /**stupid utility function for drawing particles form a zero mean, sigma variance normal distribution
 probably it should not go there*/
-double sampleGaussian(double sigma,unsigned long int S=0);
+double sampleGaussian(double sigma,unsigned int S=0);
 
 double evalGaussian(double sigmaSquare, double delta);
 double evalLogGaussian(double sigmaSquare, double delta);
diff --git a/log/carmenconfiguration.cpp b/log/carmenconfiguration.cpp
index 7cac05e..6a91404 100644
--- a/log/carmenconfiguration.cpp
+++ b/log/carmenconfiguration.cpp
@@ -10,6 +10,10 @@
 
 #define LINEBUFFER_SIZE 10000
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 namespace GMapping {
 
 using namespace std;
diff --git a/scanmatcher/scanmatch_test.cpp b/scanmatcher/scanmatch_test.cpp
index a9a3ccc..a71c1e5 100644
--- a/scanmatcher/scanmatch_test.cpp
+++ b/scanmatcher/scanmatch_test.cpp
@@ -11,6 +11,10 @@
 using namespace std;
 using namespace GMapping;
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 #define DEBUG cout << __PRETTY_FUNCTION__
 #define MAX_STRING_LENGTH 1024
 
diff --git a/scanmatcher/scanmatcher.cpp b/scanmatcher/scanmatcher.cpp
index 5d7f291..da388f4 100644
--- a/scanmatcher/scanmatcher.cpp
+++ b/scanmatcher/scanmatcher.cpp
@@ -7,11 +7,15 @@
 #include "gmapping/scanmatcher/gridlinetraversal.h"
 //#define GENERATE_MAPS
 
+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCDNAME__
+#endif
+
 namespace GMapping {
 
 using namespace std;
 
-const double ScanMatcher::nullLikelihood=-.5;
+//const double ScanMatcher::nullLikelihood=-.5;
 
 ScanMatcher::ScanMatcher(): m_laserPose(0,0,0){
 	//m_laserAngles=0;
diff --git a/utils/stat.cpp b/utils/stat.cpp
index 30536c4..7b569e9 100644
--- a/utils/stat.cpp
+++ b/utils/stat.cpp
@@ -5,6 +5,7 @@
 //#include <gsl/gsl_eigen.h>
 //#include <gsl/gsl_blas.h>
 #include <math.h>
+#include "gmapping/utils/gvalues.h"
 #include "gmapping/utils/stat.h"
 
 namespace GMapping {
@@ -45,7 +46,7 @@ double pf_ran_gaussian(double sigma)
   return(sigma * x2 * sqrt(-2.0*log(w)/w));
 }
 
-double sampleGaussian(double sigma, unsigned long int S) {
+double sampleGaussian(double sigma, unsigned int S) {
   /*
 	static gsl_rng * r = NULL;
 	if(r==NULL) {
@@ -56,7 +57,7 @@ double sampleGaussian(double sigma, unsigned long int S) {
 	if (S!=0)
         {
 		//gsl_rng_set(r, S);
-                srand48(S);
+                srand(S);
         }
 	if (sigma==0)
 		return 0;
