diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3563831..7d139fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,19 +32,19 @@ include_directories(${LIBRARY_INCLUDE_DIR} ${catkin_INCLUDE_DIRS})
 
 ## Declare libraries
 add_library(liburg_c ${LIBRARY_SRC_DIR}/urg_sensor.c
-	                 ${LIBRARY_SRC_DIR}/urg_utils.c
-	                 ${LIBRARY_SRC_DIR}/urg_debug.c
-	                 ${LIBRARY_SRC_DIR}/urg_connection.c
-	                 ${LIBRARY_SRC_DIR}/urg_ring_buffer.c
-	                 ${LIBRARY_SRC_DIR}/urg_serial.c
-	                 ${LIBRARY_SRC_DIR}/urg_serial_utils.c
-	                 ${LIBRARY_SRC_DIR}/urg_tcpclient.c
-	                 ${LIBRARY_SRC_DIR}/urg_time.c
-
+                 ${LIBRARY_SRC_DIR}/urg_utils.c
+                 ${LIBRARY_SRC_DIR}/urg_debug.c
+                 ${LIBRARY_SRC_DIR}/urg_connection.c
+                 ${LIBRARY_SRC_DIR}/urg_ring_buffer.c
+                 ${LIBRARY_SRC_DIR}/urg_serial.c
+                 ${LIBRARY_SRC_DIR}/urg_serial_utils.c
+                 ${LIBRARY_SRC_DIR}/urg_tcpclient.c
+                 ${LIBRARY_SRC_DIR}/urg_time.c
 )
-if(NOT ANDROID)
+
+if(NOT ANDROID AND NOT APPLE)
 	target_link_libraries(liburg_c -lrt -lm)
-endif(NOT ANDROID)
+endif(NOT ANDROID AND NOT APPLE)
 
 add_library(open_urg_sensor ${LIBRARY_SAMPLE_DIR}/open_urg_sensor.c)
 target_link_libraries(open_urg_sensor -lm liburg_c ${catkin_LIBRARIES})
