diff --git a/jackal_base/include/jackal_base/jackal_hardware.h b/jackal_base/include/jackal_base/jackal_hardware.h
index ab74fbd..cd2df83 100644
--- a/jackal_base/include/jackal_base/jackal_hardware.h
+++ b/jackal_base/include/jackal_base/jackal_hardware.h
@@ -35,6 +35,9 @@
 #define JACKAL_BASE_JACKAL_HARDWARE_H
 
 #include "boost/thread.hpp"
+#ifdef ERROR
+#undef ERROR
+#endif
 #include "hardware_interface/joint_state_interface.h"
 #include "hardware_interface/joint_command_interface.h"
 #include "hardware_interface/robot_hw.h"
