   /// @param state the state to set
diff --git a/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h b/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
index 99f9ab863..69ce6a7c1 100644
--- a/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
+++ b/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
@@ -45,6 +45,20 @@
 #include <ros/ros.h>
 #endif
 
+#include <ros/macros.h>
+
+// Import/export for windows dll's and visibility for gcc shared libraries.
+
+#ifdef ROS_BUILD_SHARED_LIBS // ros is being built around shared libraries
+  #ifdef moveit_planning_scene_rviz_plugin_core_EXPORTS // we are building a shared lib/dll
+    #define MOVEIT_PLANNING_SCENE_RVIZ_PLUGIN_CORE_DECL ROS_HELPER_EXPORT
+  #else // we are using shared lib/dll
+    #define MOVEIT_PLANNING_SCENE_RVIZ_PLUGIN_CORE_DECL ROS_HELPER_IMPORT
+  #endif
+#else // ros is being built around static libraries
+  #define MOVEIT_PLANNING_SCENE_RVIZ_PLUGIN_CORE_DECL
+#endif
+
 namespace Ogre
 {
 class SceneNode;
@@ -64,7 +78,7 @@ class EnumProperty;
 
 namespace moveit_rviz_plugin
 {
-class PlanningSceneDisplay : public rviz::Display
+class MOVEIT_PLANNING_SCENE_RVIZ_PLUGIN_CORE_DECL PlanningSceneDisplay : public rviz::Display
 {
   Q_OBJECT
 
