diff --git a/include/moveit_servo/servo_calcs.h b/include/moveit_servo/servo_calcs.h
index c67afb6e0..5d2488284 100644
--- a/include/moveit_servo/servo_calcs.h
+++ b/include/moveit_servo/servo_calcs.h
@@ -265,7 +265,7 @@ private:
 
   const int gazebo_redundant_message_count_ = 30;
 
-  uint num_joints_;
+  std::size_t num_joints_;
 
   // True -> allow drift in this dimension. In the command frame. [x, y, z, roll, pitch, yaw]
   std::array<bool, 6> drift_dimensions_ = { { false, false, false, false, false, false } };
diff --git a/src/cpp_interface_example/cpp_interface_example.cpp b/src/cpp_interface_example/cpp_interface_example.cpp
index 53e919ee3..028fb2fc0 100644
--- a/src/cpp_interface_example/cpp_interface_example.cpp
+++ b/src/cpp_interface_example/cpp_interface_example.cpp
@@ -108,7 +108,7 @@ int main(int argc, char** argv)
   auto joint_servo_pub = nh.advertise<control_msgs::JointJog>(servo.getParameters().joint_command_in_topic, 1);
 
   ros::Rate cmd_rate(100);
-  uint num_commands = 0;
+  std::size_t num_commands = 0;
 
   // Send a few Cartesian velocity commands
   while (ros::ok() && num_commands < 200)
